fsharp-backend/src/BackendOnlyStdLib/LibHttpClient5.fs

```diff
@@ -39,23 +36,45 @@ let parametersNoBody =
     Param.make "headers" (TDict TStr) "" ]
 
 
-let guessContentType (body : Dval option) : ContentType.T =
+// Header utility functions, deliberately kept separate from the Http
+// Middleware, as we want to be able to change them separately.
+
+let hasFormHeader (headers : HttpHeaders.T) : bool =
+  headers
+  |> HttpHeaders.get "content-type"
+  |> Option.map Tablecloth.String.toLowercase = Some
+                                                  "application/x-www-form-urlencoded"
+
+let hasJsonHeader (headers : HttpHeaders.T) : bool =
+  // CLEANUP: don't use contains for this
```

Comment:
I'm curious what's wrong with Contains here?

Reply:
It's just that it's not super strict: it could be `my-application/jsonry`